File: laravel_api/app/Models/Menu.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Menu extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = ['text', 'restaurant_id'];

    public function restaurant()
    {
        return $this->belongsTo(Restaurant::class);
    }

    public function languages()
    {
        return $this->belongsToMany(Language::class);
    }

    public function translation()
    {
        return $this->hasMany(Translation::class);
    }
}

File: menu_flutter/lib/models/app/menu.dart
class Menu {
  Menu({required this.id, required this.text, required this.restaurantId});

  int id;
  String text;
  int restaurantId;

  static Menu fromJson(dynamic obj) {
    return Menu(
      id: obj['id'],
      text: obj['text'],
      restaurantId: obj['restaurant_id']
    );
  }
}
